pkgs/development/interpreters/lua-5/build-lua-package.nix

```diff
@@ -106,15 +93,30 @@ let
     );
   externalDeps' = lib.filter (dep: !lib.isDerivation dep) externalDeps;
 
-  luarocksDrv = luaLib.toLuaModule ( lua.stdenv.mkDerivation (
-builtins.removeAttrs attrs ["disabled" "checkInputs" "externalDeps" "extraVariables"] // {
-
-  name = namePrefix + pname + "-" + version;
+  luarocksDrv = luaLib.toLuaModule ( lua.stdenv.mkDerivation (self: let
+
+    rocksSubdir = "${self.pname}-${self.version}-rocks";
+    luarocks_content = let
+      generatedConfig = luaLib.generateLuarocksConfig {
+        externalDeps = externalDeps ++ externalDepsGenerated;
+        inherit extraVariables;
+        inherit rocksSubdir;
+        inherit requiredLuaRocks;
+      };
+      in
+        ''
+        ${generatedConfig}
+        ${extraConfig}
+        '';
```

Comment:
```suggestion
    generatedConfig = luaLib.generateLuarocksConfig {
      externalDeps = externalDeps ++ externalDepsGenerated;
      inherit extraVariables requiredLuaRocks rocksSubdir;
    };
    luarocks_content = ''
      ${generatedConfig}
      ${extraConfig}
    '';
```

Reply:
I will incorporate your suggestion and repush later today to see if it fixes the manual generation check